libs/auth/src/java_card.ts

```diff
@@ -438,19 +455,33 @@ export class JavaCard implements Card {
    *
    * This wrapper should never throw errors.
    */
-  private async safeReadCardDetails(): Promise<Optional<CardDetails>> {
+  private async safeReadCardDetails(): Promise<CardDetails> {
     try {
       return await this.readCardDetails();
-    } catch {
-      return undefined;
+    } catch (error) {
+      const errorMessage = extractErrorMessage(error);
+      const reason = (() => {
+        if (errorMessage.includes('certificate has expired')) {
+          return 'certificate_expired';
+        }
+        /* istanbul ignore next: It's hard to create test certs with start dates in the future
```

Comment:
Could we create one that isn't valid for 100 years?

Reply:
So that won't quite do it because that'll just bring up the cert end date, when what we want is to push out the cert start date

I do in fact create a cert that is valid for a very short duration to test the `certificate_expired` case: https://github.com/votingworks/vxsuite/blob/3a84241d7a70c99c1fb63d3b7d339fa00ff82b42/libs/auth/scripts/src/generate_dev_keys_and_certs.ts#L288-L292

Reply:
Sorry, restating my comment above more clearly:

Could we create a certificate whose validity window doesn't start until 100 years from now?

Reply:
Ah yup, so I posted some thinking on creating certs with a start date in the future above:

> there are ways to do it to be clear, but they require either messing around with the system time using sudo, which doesn't lend itself nicely to a script that can be used by others, or switching the cert creation logic from openssl x509 to openssl ca, which I don't want to do this late in the game. But would like to update post cert freeze!

`openssl x509 -req` always sets the cert start time to the current time.
`openssl ca -in` supports a `-startdate` param, but I don't want to make that switch right before cert.

The idea of using `sudo` to temporarily change the time isn't terrible, but also requires you to toggle some settings around time syncing.

I could also make such a cert in a one-off capacity, but it would then be hard for others to regenerate it should the need ever arise. It'd be more like a manual fixture. Which is maybe fine if I label it as such, but I'm not a huge fan of that idea.

Reply:
Okay, that makes sense. This PR is really strongly making me feel that using the `openssl` binary directly is _not_ a great idea. I understand the FIPS requirement, but I think this should be on the list of things to consider changing in the future. For now, this is fine.

Reply:
Yup I totally agree! This PR convinced me of that as well. Would be great to have proper error codes.

I wanted to get these updated error screens into this release because they come up so often but wouldn't want us to continue down the route of conditioning on error message strings for too long.

Funnily enough, there's discussion on changing the OpenSSL error message for this case to clue users into clock mismatches 😄 https://github.com/openssl/openssl/issues/14771